.evergreen/config.yml

```diff
@@ -0,0 +1,176 @@
+      params:
+        script: |
+          cat driver-src/driver-expansion.yml
+    - command: expansions.update
+      params:
+        file: driver-src/driver-expansion.yml
+    # Run driver-supplied script to install the driver.
+    - command: shell.exec
+      params:
+        working_dir: driver-src
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          ${PREPARE_SHELL_DRIVER}
+          sh .evergreen/atlas-testing/install-driver.sh
+
+  "run tests":
+    # Run the Atlas Planned Maintenance Tests.
+    - command: shell.exec
+      type: test
+      params:
+        script: |
+          ${PREPARE_SHELL_COMMON}
+          ${PREPARE_SHELL_DRIVER}
+          "$ASTROLABE_BINARY" spec-tests run astrolabe-src/tests -e driver-src/.evergreen/atlas-testing/workload-executor.sh
+
+  "upload test results":
+    # Upload the xunit-format test results.
+    - command: attach.xunit_results
+      params:
+        file: "xunit-output/*.xml"
+
+# Functions to run before the test.
+pre:
+  - func: "install astrolabe"
+  - func: "install driver"
+
+# Functions to run after the test.
+post:
+  - func: "upload test results"
+
+tasks:
+    # Generic task to be used by all drivers.
+    - name: "run-tests"
+      commands:
+        - func: "run tests"
+
+axes:
+  # The 'driver' axis specifies the driver to be tested (including driver version).
+  # Drivers MUST specify the DRIVER_REPOSITORY and DRIVER_REVISION variables.
+  - id: driver
+    display_name: driver
+    values:
+      - id: pymongo-master
+        display_name: "PyMongo (master)"
+        variables:
+          DRIVER_REPOSITORY: "https://github.com/prashantmital/mongo-python-driver.git"
+          DRIVER_REVISION: "PYTHON-1743/test-atlas-planned-maintenance"
+
+  # The 'platform' axis specifies the evergreen host distro to use.
+  # Platforms MUST specify the SYSTEM_PYTHON3_BINARY variable as this is required to install astrolabe.
+  - id: platform
+    display_name: OS
+    values:
+      - id: ubuntu-16.04
+        display_name: "Ubuntu 16.04"
+        run_on: ubuntu1604-test
+        batchtime: 10080  # 7 days
+        variables:
+          SYSTEM_PYTHON3_BINARY: "/opt/python/3.7/bin/python3"
+
+  # The 'runtime' axis specifies the language runtime to use.
+  # Runtimes SHOULD specify language and platform-specific variables needed by driver scripts.
+  - id: runtime
+    display_name: runtime
+    values:
+      - id: python36
+        display_name: CPython-3.6
+        variables:
+          PYTHON_BINARY: "/opt/python/3.6/bin/python3"
+      - id: python37
+        display_name: CPython-3.7
+        variables:
+          PYTHON_BINARY: "/opt/python/3.7/bin/python3"
+      - id: python38
+        display_name: CPython-3.8
+        variables:
+          PYTHON_BINARY: "/opt/python/3.8/bin/python3"
+
+buildvariants:
+- matrix_name: "tests-python"
+  matrix_spec:
+    driver: ["pymongo-master"]
+    platform: ["ubuntu-16.04"]
+    runtime: ["python36", "python37", "python38"]
```

Comment:
We should add 2.7 too. If there is some complexity adding 2.7 we should open a follow up ticket for it.

Reply:
Will do. I don't think there is any complexity associated with adding that but it wasn't added here because the framework itself is Py3-only and I got into the bad habit of installing PyMongo into the same env as astrolabe during testing.

Reply:
Yeah, doesn't astrolabe itself use pymongo? So the versions could clash?

Reply:
In the way the evergreen job is setup, this is a non-issue as I am using different virtualenvs for astrolabe and pymongo. When testing locally, that could certainly happen but it wasn't a major concern in the prototyping phase.

Reply:
I think you forgot to add Python 2.7 here.

Reply:
Done.